llvm/lib/Target/LoongArch/LoongArchISelLowering.cpp

```diff
@@ -2855,6 +2855,10 @@ LoongArchTargetLowering::lowerEXTRACT_VECTOR_ELT(SDValue Op,
   case MVT::v16i16:
   case MVT::v4i64:
   case MVT::v4f64: {
+    // TODO: Similar optimization can be applied for la32.
+    if (!Subtarget.is64Bit())
+      return SDValue();
+
```

Comment:
Line 2871: `DAG.getNode(LoongArchISD::MOVGR2FR_W_LA64, DL, MVT::f32, Idx);`

```suggestion
Subtarget.is64Bit()
    ? DAG.getNode(LoongArchISD::MOVGR2FR_W_LA64, DL, MVT::f32, Idx)
    : DAG.getBitcast(MVT::f32, Idx);
```

Reply:
Great suggestions. I will modify it later.